include/seqan3/range/views/minimiser.hpp

```diff
@@ -0,0 +1,481 @@
+        if (window_right == urange_end)
+            return true;
+
+        uint64_t new_value = *window_right;
+        if (minimiser_value == *(std::begin(window_values)))
+        {
+            window_values.pop_front();
+            if (!window_values.empty())
+            {
+                window_values.push_back(new_value);
+                minimiser_value = *(std::min_element(std::begin(window_values), std::end(window_values)));
+                return true;
+            }
+
+        }
+        else
+        {
+            window_values.pop_front();
+        }
+
+        window_values.push_back(new_value);
+
+        if (new_value < minimiser_value)
+        {
+            minimiser_value = new_value;
+            return true;
+        }
+        else if (window_values.size() == 1)
+        {
+            return true;
+        }
+
+        return false;
+    }
+};
+
+//!\brief A deduction guide for the view class template.
+template <std::ranges::viewable_range rng_t>
+minimiser_view(rng_t &&, uint32_t const & window_values_size) -> minimiser_view<std::ranges::all_view<rng_t>>;
+
+
+// ---------------------------------------------------------------------------------------------------------------------
+// minimiser_fn (adaptor definition)
+// ---------------------------------------------------------------------------------------------------------------------
+
+//![adaptor_def]
+//!\brief views::minimiser's range adaptor object type (non-closure).
+struct minimiser_fn
+{
+    //!\brief Store the number of values in one window and return a range adaptor closure object.
+    constexpr auto operator()(uint32_t const & window_values_size) const
+    {
+        return adaptor_from_functor{*this, window_values_size};
+    }
+
+    /*!\brief                           Call the view's constructor with two arguments: the underlying view and an
+     *                                  integer indicating how many values one window contains.
+     * \param[in] urange                The input range to process. Must model std::ranges::viewable_range and
+     *                                  std::ranges::forward_range.
+     * \param[in] window_values_size   The number of values in one window.
+     * \returns                         A range of converted values.
+     */
+    template <std::ranges::range urng_t>
+    constexpr auto operator()(urng_t && urange, uint32_t const & window_values_size) const
+    {
+        static_assert(std::ranges::viewable_range<urng_t>,
+            "The range parameter to views::minimiser cannot be a temporary of a non-view range.");
+        static_assert(std::ranges::forward_range<urng_t>,
+            "The range parameter to views::minimiser must model std::ranges::forward_range.");
+
+        return minimiser_view{urange, window_values_size};
+    }
+};
+//![adaptor_def]
+
+} // namespace seqan3::detail
+
+namespace seqan3::views
+{
+
+/*!\name Alphabet related views
```

Comment:
Is this an alphabet related view?

Reply:
No, what would be better? Comparable views?

Reply:
I just checked. We only have two categories so far. I guess it's `General purpose views` then